### DISCUSSION TITLE
Problema al intentar correr los tests

### DISCUSSION THREAD

**Discussion by antoopuelma** [IIC2133-PUC/2024-2#115] [Tarea 1] [answered]
Hola! Tengo un problema cuando intento correr el comando ./happy happy/easy/input.txt out.txt, ya que me aparece un segmentation fault. Yo tengo guardada en el mismo lugar la carpeta de los tests happy y la carpeta de mi código, y no se si el error sale porque estoy poniendo mal la ruta o si es porque estoy trabajando de un mac (intenté hacer make start pero me sale error, y sin eso no sé como acceder al docker)

Me serviría mucho si pudieran ayudarme

**Comment by tomastrivino** [comment, marked as answer]
Asegúrate de poner bien la ruta. Si el segmentation fault sale de inmediato podría deberse a un problema con eso.

**Reply by antoopuelma** [reply]
que yo tengo mi carpeta happy (donde se encuentran los tests de la parte happy) y mi carpeta T1 en el mismo lugar. según yo la sintaxis que puse para correr el primer test es la correcta ./happy happy/easy/input.txt out.txt (ya que así lo corría para la t0) 
sino como debería ser?

**Reply by tomastrivino** [reply]
¿Qué output te da el comando `ls happy/easy`? Debería listar los contenidos de dicha carpeta. Ejecuta este comando desde la misma carpeta en que corre el ejecutable.

**Reply by antoopuelma** [reply]
me sale esto: ls: happy/easy: Not a directory

**Reply by tomastrivino** [reply]
Hola,
eso significa que no estás poniendo el path adecuadamente. Si estás en la carpeta en donde ejecutas `./happy`, ejecuta `ls` para mostrar los archivos y carpetas que hay desde el path en que te encuentras. Eso podría ayudarte.

Saludos

**Reply by antoopuelma** [reply]
hice ls y ahora ya estoy en la carpeta happy (de src), que es donde tengo mis archivos, pero al correr este comando:./happy happy/easy/input.txt out.txt, me sale este error: zsh: no such file or directory: ./happy 
que estoy haciendo mal?

**Reply by tomastrivino** [reply]
Supongamos que tienes la siguiente estructura de carpetas
```
[Carpeta T1]
- src
-- happy
-- order
- test
-- happy
-- order
- happy (ejecutable)
- order (ejecutable)
```
Debes estar en la carpeta en donde se guarda el ejecutable que se compiló. Asumo por lo que me comentas que eso se guardó en la carpeta T1, la que además contiene otras carpetas como src.

Lo que te está diciendo el error es que estás intentando correr el ejecutable (happy) pero simplemente no existe en la carpeta en la que te ubicas actualmente. Asegúrate de ejecutar ese comando desde la carpeta correspondiente.

**Reply by antoopuelma** [reply]
estoy en el caso que me comentas de estructura, ahora abrí la terminal en mi carpeta T1, y corrí ./happy happy/easy/input.txt out.txt, apareciendo nuevamente el error de segmentation, en cual carpeta debo estar para correr el ejecutable?

**Reply by tomastrivino** [reply]
Estás en la carpeta correcta para correr el ejecutable. ¡Puede ser que el otro path, `happy/easy/input.txt` esté mal!

**Reply by antoopuelma** [reply]
acabo de intentar con este path: ./happy tests/happy/easy/input.txt out.txt, pero me sale este error: zsh: segmentation fault  ./happy tests/happy/easy/input.txt out.txt, como puedo arreglarlo?

**Reply by tomastrivino** [reply]
Si es que el archivo del path (`tests/happy/easy/input.txt `) que pusiste realmente existe, entonces ya correspondería a un problema a nivel código.
